LCLSGeneral/LCLSGeneral/POUs/Hardware/FB_TempSensor.TcPOU

```diff
@@ -0,0 +1,51 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <POU Name="FB_TempSensor" Id="{d7de6be9-251a-4df0-a4bb-4330b01e3329}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_TempSensor
+(*
+    Handles scaling and default diagnostics for temperature sensors,
+    such as thermocouples, RTDs, and others.
+    2020-03-02 Zachary Lentz
+*)
+VAR_INPUT
+    // Ratio between raw value and actual temperature. Default is 10 for 10 steps per degree (or 0.1 degree resolution)
+    iScale: INT := 10;
+END_VAR
+VAR_OUTPUT
+    {attribute 'pytmc' := '
+        pv: TEMP
+        io: input
+        field: EGU C
+        field: PREC 2
+    '}
+    fTemp: LREAL;
+
+    {attribute 'pytmc' := '
+        pv: CONN
+        io: input
+        field: ONAM Connected
+        field: ZNAM Disconnected
+    '}
+    bConnected: BOOL;
+
+    {attribute 'pytmc' := '
+        pv: ERR
+        io: input
+        field: ONAM True
+        field: ZNAM False
+    '}
+    bError AT %I*: BOOL;
```

Comment:
It shows my relative PLC newbie-ness here, but it's odd to me to have a `VAR_OUTPUT` block where `%I*`s are declared - but this just allows for some linking specification I suppose?

I wonder if there's a way we can verify all required links have been connected.

Reply:
Ah yeah this is confusing, it's definitely just a link point. The terminal has an error bit that should be linked here so that EPICS and other things in the PLC program can check the error state. It was originally just a `VAR` but I decided it might be useful in the PLC.

The only thing I can think of for verification is to initialize it to a bad value, e.g. I can make this `bError AT %I*: BOOL := TRUE` and then if you forget to link it you will always be in the error state.